src/znc.cpp

```diff
@@ -1738,12 +1748,21 @@ bool CZNC::AddListener(CConfig* pConfig, CString& sError) {
 #endif
     bool bIRC;
     bool bWeb;
+    CString sPort;
     unsigned short uPort;
-    if (!pConfig->FindUShortEntry("port", uPort)) {
+    unsigned long int ulPort; //read the port with some extra bits for validation
+    if (!pConfig->FindStringEntry("port", sPort)) {
         sError = "No port given";
         CUtils::PrintError(sError);
         return false;
     }
+    ulPort = sPort.ToULong();
+    if (ulPort > 65535 || ulPort < 1) {
+	sError = t_f("Port [{1}] is out of the 1-65535 range")(sPort);
```

Comment:
could you run clang-format?

Reply:
@DarthGandalf sure, I'll run, my fault, I'm note used to writing code anymore... Any project specific usage? Or just standard?

Reply:
It's already configured via https://github.com/znc/znc/blob/master/.clang-format so either just run it from command line, or if your IDE has some way to invoke it for the changed lines only, that would be best

(unfortunately clang-format has/had bugs in how it handles tabs :( )